tests/integration/test_wpk/test_wpk_agent.py

```diff
@@ -358,8 +376,36 @@ def test_wpk_agent(get_configuration, prepare_agent_version, download_wpk,
             upgrade_exec_message = str(exp_json['message'])
         assert upgrade_exec_message == expected['error_message'], \
                f'Expected error message does not match'
+
     if upgrade_process_result and expected['receive_notification']:
-        result = remoted_simulator.wait_upgrade_notification(timeout=180)
+        if sys_platform not in ['win32', 'Windows']:
+            lines = count_file_lines(tools.LOG_FILE_PATH) 
+            while lines != 0:
+                time.sleep(1) 
+                lines = count_file_lines(tools.LOG_FILE_PATH) 
+        else:
+            truncate_file(tools.LOG_FILE_PATH)
+        wazuh_log_monitor = FileMonitor(tools.LOG_FILE_PATH)
+
+
+        if metadata['simulate_rollback']:
+
+            if sys_platform not in ['win32', 'Windows']:
+                wazuh_log_monitor.start(timeout=200,
+                                        error_message="Error agentd not stopped",
+                                        callback=callback_agent_stop())
+ 
+
+            wazuh_log_monitor.start(timeout=200,
+                                    error_message="Upgrade module did not start",
+                                    callback=callback_upgrade_module_up())
+
+                                    
+            remoted_simulator.change_default_listener = True
+
+        result = json.loads(wazuh_log_monitor.start(timeout=300, error_message="ACK event not received", callback=callback_detect_upgrade_ack_event).result())['parameters']
```

Comment:
This line is a little hard to read, could we split it in 2?
```suggestion
        event = wazuh_log_monitor.start(timeout=300, error_message="ACK event not received", callback=callback_detect_upgrade_ack_event).result()
        result = json.loads(event)['parameters']
```

Reply:
Done 29484935443c9d15880a267f5abcd72f8779aa3a

Reply:
I'm still not sold on these lines, if `callback_detect_upgrade_ack_event` is supposed to always capture a json string we can move the `json.loads` operation inside the callback and have it return a dictionary directly. You can look into `callback_detect_event` in fim.py were a similar situation to this one is shown.

Reply:
Done dd346a586a5a232163c807a905dc7b35d4b7aa4e